examples/vision/gradient_centralization.py

```diff
@@ -0,0 +1,253 @@
+from tensorflow.keras import layers
+from tensorflow.keras.layers import Conv2D
+from tensorflow.keras.layers import MaxPooling2D
+from tensorflow.keras.layers import Dropout
+from tensorflow.keras.layers import Dense
+from tensorflow.keras.optimizers import RMSprop
+
+
+import gctf
+
+from time import time
+import os
+
+"""
+# Prepare the data
+
+For this example, we will be using the [Horses or Humans
+dataset](https://www.tensorflow.org/datasets/catalog/horses_or_humans).
+"""
+
+num_classes = 2
+input_shape = (300, 300, 3)
+dataset_name = "horses_or_humans"
+
+(train_ds, test_ds), metadata = tfds.load(
+    name=dataset_name,
+    split=[tfds.Split.TRAIN, tfds.Split.TEST],
+    with_info=True,
+    as_supervised=True,
+)
+
+print(f"Image shape: {metadata.features['image'].shape}")
+print(f"Training images: {metadata.splits['train'].num_examples}")
+print(f"Test images: {metadata.splits['test'].num_examples}")
+
+"""
+# Use Data Augmentation
+
+We will rescale the data to `[0, 1]`  andperform simple augmentations to our data.
+"""
+
+rescale = tf.keras.Sequential([layers.experimental.preprocessing.Rescaling(1.0 / 255)])
+
+data_augmentation = tf.keras.Sequential(
+    [
+        layers.experimental.preprocessing.RandomFlip("horizontal_and_vertical"),
+        layers.experimental.preprocessing.RandomRotation(0.3),
+        layers.experimental.preprocessing.RandomZoom(0.2),
+    ]
+)
+
+batch_size = 128
+AUTOTUNE = tf.data.AUTOTUNE
+
+
+def prepare(ds, shuffle=False, augment=False):
+    # Rescale dataset
+    ds = ds.map(lambda x, y: (rescale(x), y), num_parallel_calls=AUTOTUNE)
+
+    if shuffle:
+        ds = ds.shuffle(1024)
+
+    # Batch dataset
+    ds = ds.batch(batch_size)
+
+    # Use data augmentation only on the training set
+    if augment:
+        ds = ds.map(
+            lambda x, y: (data_augmentation(x, training=True), y),
+            num_parallel_calls=AUTOTUNE,
+        )
+
+    # Use buffered prefecting
+    return ds.prefetch(buffer_size=AUTOTUNE)
+
+
+"""
+Rescale and augment the data
+"""
+
+train_ds = prepare(train_ds, shuffle=True, augment=True)
+test_ds = prepare(test_ds)
+
+"""
+# Define a model
+
+In this section we will define a Convolutional neural network
+"""
+
+model = tf.keras.Sequential(
+    [
+        layers.Conv2D(16, (3, 3), activation="relu", input_shape=(300, 300, 3)),
+        layers.MaxPooling2D(2, 2),
+        layers.Conv2D(32, (3, 3), activation="relu"),
+        layers.Dropout(0.5),
+        layers.MaxPooling2D(2, 2),
+        layers.Conv2D(64, (3, 3), activation="relu"),
+        layers.Dropout(0.5),
+        layers.MaxPooling2D(2, 2),
+        layers.Conv2D(64, (3, 3), activation="relu"),
+        layers.MaxPooling2D(2, 2),
+        layers.Conv2D(64, (3, 3), activation="relu"),
+        layers.MaxPooling2D(2, 2),
+        layers.Flatten(),
+        layers.Dropout(0.5),
+        layers.Dense(512, activation="relu"),
+        layers.Dense(1, activation="sigmoid"),
+    ]
+)
```

Comment:
Please print a summary. Also please let me know what the summary looks like (do you have a live notebook?)

Reply:
Added that. By "a live notebook" do you mean something like a runnable Colab? If yes here it is [Colab Notebook](https://colab.research.google.com/drive/1S68JYKnhJs69s1sptkYqVqk8N8SnP8lx?usp=sharing).